### PR TITLE
XD-3354 Setup Shell integration tests

### PR DESCRIPTION
**Pull request by ilayaperumalg**
- Setup Shell integration tests infrastructure
   - Moved the setup from XD to here with minimal things needed to
get started

### CODE REVIEW

File: spring-cloud-data-shell/src/test/java/org/springframework/cloud/data/shell/command/StreamCommandTests.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package org.springframework.cloud.data.shell.command;
```
Comment (markfisher) [inline spring-cloud-data-shell/src/test/java/org/springframework/cloud/data/shell/command/StreamCommandTests.java]: copyright header

**Comment by markfisher** [comment]
we should take this opportunity to favor composition over inheritance, specifically I believe the AbstractCommandTemplate base class is not worth maintaining since it only provides one method and getShell().executeCommand() is already available.

File: spring-cloud-data-shell/src/test/java/org/springframework/cloud/data/shell/AbstractShellIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+	@BeforeClass
+	public static synchronized void startUp() throws InterruptedException, IOException {
+		if (application == null) {
+			application = new AdminApplication();
+			application.main(new String[] {
+					String.format("--server.port=%s", adminPort),
+					"--spring.main.show_banner=false"});
+		}
+		shell = new Bootstrap(new String[] {"--port", String.valueOf(adminPort)}).getJLineShellComponent();
```
Comment (markfisher) [inline spring-cloud-data-shell/src/test/java/org/springframework/cloud/data/shell/AbstractShellIntegrationTest.java]: It's worth adding a delegating stop() in the CloudDataShell, thus no need to have both the shell and cloudDataShell fields? (could just have the CloudDataShell assigned to a 'shell' field instead)

Comment (ilayaperumalg) [inline spring-cloud-data-shell/src/test/java/org/springframework/cloud/data/shell/AbstractShellIntegrationTest.java]: Yeah, good idea. trying it.

**Comment by markfisher** [comment]
rebased, squashed, and merged as https://github.com/spring-cloud/spring-cloud-data/commit/41b40079c85bed5881de31f1f67cdba3cf0c802d

**Pull request closed without merging**